validator/accounts/status.go

```diff
@@ -0,0 +1,176 @@
+		select {
+		case statuses := <-statusChannel:
+			allStatuses = append(allStatuses, statuses)
+		case e := <-errorChannel:
+			err = e
+		}
+	}
+
+	return allStatuses, err
+}
+
+func fetchValidatorStatus(
+	ctx context.Context,
+	rpcProvder ethpb.BeaconNodeValidatorClient,
+	pubkeys [][]byte,
+	statusChannel chan []ValidatorStatusMetadata,
+	errorChannel chan error) {
+	if ctx.Err() == context.Canceled {
+		errorChannel <- errors.Wrap(ctx.Err(), "context has been canceled.")
+		return
+	}
+
+	req := &ethpb.MultipleValidatorStatusRequest{PublicKeys: pubkeys}
+	resp, err := rpcProvder.MultipleValidatorStatus(ctx, req)
+	if err != nil {
+		errorChannel <- errors.Wrap(
+			err,
+			fmt.Sprintf("Failed to fetch validator statuses for %d key(s)", len(pubkeys)))
+		return
+	}
+
+	statusChannel <- responseToSortedMetadata(resp)
+}
+
+func responseToSortedMetadata(resp *ethpb.MultipleValidatorStatusResponse) []ValidatorStatusMetadata {
+	pubkeys := resp.GetPublicKeys()
+	validatorStatuses := make([]ValidatorStatusMetadata, len(pubkeys))
+	for i, status := range resp.GetStatuses() {
+		validatorStatuses[i] = ValidatorStatusMetadata{
+			PublicKey: pubkeys[i],
+			Metadata:  status,
+		}
+	}
+	sort.Slice(validatorStatuses, func(i, j int) bool {
+		return validatorStatuses[i].Metadata.Status < validatorStatuses[j].Metadata.Status
+	})
+	return validatorStatuses
+}
+
+// ExtractPublicKeys extracts only the public keys from the decrypted keys from the keystore.
+func ExtractPublicKeys(decryptedKeys map[string]*keystore.Key) [][]byte {
+	i := 0
+	pubkeys := make([][]byte, len(decryptedKeys))
+	for _, key := range decryptedKeys {
+		pubkeys[i] = key.PublicKey.Marshal()
+		i++
+	}
+	return pubkeys
+}
+
+// MergeStatuses merges k sorted ValidatorStatusMetadata slices to 1.
+func MergeStatuses(allStatuses [][]ValidatorStatusMetadata) []ValidatorStatusMetadata {
+	if len(allStatuses) == 0 {
+		return []ValidatorStatusMetadata{}
+	}
+	if len(allStatuses) == 1 {
+		return allStatuses[0]
+	}
+	leftHalf := allStatuses[:len(allStatuses)/2]
+	rightHalf := allStatuses[len(allStatuses)/2:]
+	return mergeTwo(MergeStatuses(leftHalf), MergeStatuses(rightHalf))
+}
+
+func mergeTwo(s1, s2 []ValidatorStatusMetadata) []ValidatorStatusMetadata {
+	i, j, k := 0, 0, 0
+	sortedStatuses := make([]ValidatorStatusMetadata, len(s1)+len(s2))
+	for j < len(s1) && k < len(s2) {
+		if s1[j].Metadata.Status < s2[k].Metadata.Status {
+			sortedStatuses[i] = s1[j]
+			j++
+		} else {
+			sortedStatuses[i] = s2[k]
+			k++
+		}
+		i++
+	}
+	for j < len(s1) {
+		sortedStatuses[i] = s1[j]
+		i, j = i+1, j+1
+	}
+	for k < len(s2) {
+		sortedStatuses[i] = s2[k]
+		i, k = i+1, k+1
+	}
+	return sortedStatuses
+}
+
+// PrintValidatorStatusMetadata prints out validator statuses and its corresponding metadata
+func PrintValidatorStatusMetadata(validatorStatuses []ValidatorStatusMetadata) {
+	for _, v := range validatorStatuses {
+		m := v.Metadata
+		key := v.PublicKey
+		fmt.Printf(
```

Comment:
Can we use logrus here for consistency? log.WithField or log.Infof

Reply:
https://github.com/prysmaticlabs/prysm/pull/5784/commits/de2f7a89158bc3f83f084ae29b0270a15eddbcaa